python_modules/dagster-graphql/dagster_graphql/implementation/fetch_assets.py

```diff
@@ -311,17 +315,23 @@ def get_materialization_cts_grouped_by_dimension(
         [b|x count, b|y count]
     ]
     """
-    primary_dimension, secondary_dimension = group_by_dimensions[0], group_by_dimensions[1]
     db_materialization_counts = get_materialization_ct_in_tags(
-        graphene_info, asset_key, group_by_dimensions
+        graphene_info, asset_key, partition_dimensions
     )
     materialization_counts_grouped_by_dimension: List[List[int]] = []
 
-    for primary_dim_key in partition_keys_by_dimension[primary_dimension]:
+    primary_dimension = partition_dimensions[0]
+    secondary_dimension = partition_dimensions[1]
+
+    for (
+        primary_dim_key
+    ) in (
+        primary_dimension.external_partitions_def_data.get_partitions_definition().get_partition_keys()
+    ):
         materialization_counts_grouped_by_dimension.append(
             [
                 db_materialization_counts.get(primary_dim_key, {}).get(secondary_dim_key, 0)
-                for secondary_dim_key in partition_keys_by_dimension[secondary_dimension]
+                for secondary_dim_key in secondary_dimension.external_partitions_def_data.get_partitions_definition().get_partition_keys()
```

Comment:
This is a dumb nit, but I'm not sure how expensive this is:
```
secondary_dimension.external_partitions_def_data.get_partitions_definition().get_partition_keys()
```

It might be nice to prefetch it into a variable outside the inner loop so it's not repeated for each primary dimension key!